math/src/polynomial.rs

```diff
@@ -1,50 +1,48 @@
-use super::field::fields::u64_prime_field::U64FieldElement;
+use super::field::element::FieldElement;
+use crate::field::traits::IsField;
 use std::ops;
 
 /// Represents the polynomial c_0 + c_1 * X + c_2 * X^2 + ... + c_n * X^n
 /// as a vector of coefficients `[c_0, c_1, ... , c_n]`
 #[derive(Debug, Clone, PartialEq, Eq)]
-pub struct Polynomial<const ORDER: u64> {
-    coefficients: Vec<U64FieldElement<ORDER>>,
+pub struct Polynomial<FE> {
+    coefficients: Vec<FE>,
 }
 
-impl<const ORDER: u64> Polynomial<ORDER> {
+impl<F: IsField> Polynomial<FieldElement<F>> {
     /// Creates a new polynomial with the given coefficients
-    pub fn new(coefficients: Vec<U64FieldElement<ORDER>>) -> Self {
+    pub fn new(coefficients: Vec<FieldElement<F>>) -> Self {
         // Removes trailing zero coefficients at the end
         let mut unpadded_coefficients = coefficients
             .into_iter()
             .rev()
-            .skip_while(|x| *x == U64FieldElement::new(0))
-            .collect::<Vec<U64FieldElement<ORDER>>>();
+            .skip_while(|x| *x == FieldElement::zero())
+            .collect::<Vec<FieldElement<F>>>();
         unpadded_coefficients.reverse();
```

Comment:
We could use [`take_while`](https://doc.rust-lang.org/std/iter/trait.Iterator.html#method.take_while) (`*x != FieldElement::zero()`) and remove the `.rev()` step and the `unpadded_coefficients.reverse();`.

Reply:
I think that would break for polynomials that have zeros in the middle of their representation.

Reply:
Yes, you're right! Let's keep it this way.